src/libNode/Node.cpp

```diff
@@ -621,21 +631,34 @@ bool Node::ProcessTxnPacketFromLookup(
     {
         // check for recieval of new ds block
         // need to wait the ProcessDSBlock finish
-        lock_guard<mutex> g(m_mutexDSBlock);
+        lock_guard<mutex> g1(m_mutexDSBlock);
```

Comment:
should we lock both g1 and g2 together at the same time. If not careful, one may introduce codes in the future that cause a deadlock. i.e hold onto g2 first then g1.

Reply:
g2 only lock when the if condition is true, as the two locks are for different purpose I still feel it's better to keep them separate